library/src/actions/jwt/jwt.ts

```diff
@@ -0,0 +1,193 @@
+import type {
+  BaseIssue,
+  BaseValidation,
+  ErrorMessage,
+} from '../../types/index.ts';
+import { _addIssue } from '../../utils/index.ts';
+
+const POSSIBLE_JWT_REGEX = /^(?:[\w-]+\.){2}[\w-]+$/u;
+
+// https://datatracker.ietf.org/doc/html/rfc7518#section-3.1
+type JwtAlgorithm =
+  | 'HS256'
+  | 'HS384'
+  | 'HS512'
+  | 'RS256'
+  | 'RS384'
+  | 'RS512'
+  | 'ES256'
+  | 'ES384'
+  | 'ES512'
+  | 'PS256'
+  | 'PS384'
+  | 'PS512'
+  | 'none';
```

Comment:
The `none` algorithm has a problem mentioned [here](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/) under the title `Meet the "None" Algorithm` but the algorithm was present in the [RFC](https://datatracker.ietf.org/doc/html/rfc7518#section-3.1). I think it we should follow the RFC as we are just checking the structure of the string. I wanted to mention it here to let you know.